src/javascript/DesignSystem/DatePicker/YearMonthSelector/YearMonthSelector.jsx

```diff
@@ -19,31 +37,48 @@ const YearMonthSelectorCmp = ({date, months, onChange}) => {
     }
 
     const handleChange = e => {
-        const {year, month} = e.target.form;
-        onChange(new Date(year.value, month.value));
+        const {name, value} = e.target;
+        if (name === 'month') {
+            onChange(new Date(selectedYear, value));
+        } else {
+            onChange(new Date(value, selectedMonth));
+        }
```

Comment:
Could be great to separate this in two different handleChange. now that we handle them outside of the form, also we do not need the name from targe if we do that.

- handleMonthChange
- handleYearChange